stdlib/xml/dom/xmlbuilder.pyi

```diff
@@ -1,6 +1,136 @@
+    errorHandler: _DOMBuilderErrorHandlerType
+    filter: DOMBuilderFilter | None  # a guess, but seems likely
+    ACTION_REPLACE: Literal[1]
+    ACTION_APPEND_AS_CHILDREN: Literal[2]
+    ACTION_INSERT_AFTER: Literal[3]
+    ACTION_INSERT_BEFORE: Literal[4]
+    _legal_actions: tuple[Literal[1], Literal[2], Literal[3], Literal[4]]
+    def __init__(self) -> None: ...
+    _options: Options
+    def setFeature(self, name: str, state: int) -> None: ...
+    def supportsFeature(self, name: str) -> bool: ...
+    def canSetFeature(self, name: str, state: int) -> bool: ...
+    _settings: dict[tuple[str, int], list[tuple[str, int]]]
+    # getFeature could return any attribute from an instance of `Options`
+    def getFeature(self, name: str) -> Any: ...
+    def parseURI(self, uri: str) -> ExpatBuilder | ExpatBuilderNS: ...
+    def parse(self, input: DOMInputSource) -> ExpatBuilder | ExpatBuilderNS: ...
+    # `input` and `cnode` argtypes for `parseWithContext` are unknowable
+    # as the function does nothing with them, and always raises an exception.
+    # But `input` is *probably* `DOMInputSource`?
+    def parseWithContext(self, input: object, cnode: object, action: Literal[1, 2, 3, 4]) -> NoReturn: ...
+    def _parse_bytestream(self, stream: BinaryIO, options: Options) -> ExpatBuilder | ExpatBuilderNS: ...
+
+def _name_xform(name: str) -> str: ...
+
+class DOMEntityResolver:
+    _opener: OpenerDirector
+    def resolveEntity(self, publicId: str | None, systemId: str) -> DOMInputSource: ...
+    def _get_opener(self) -> OpenerDirector: ...
+    def _create_opener(self) -> OpenerDirector: ...
+    def _guess_media_encoding(self, source: DOMInputSource) -> str: ...
+
+class DOMInputSource:
+    byteStream: OpenerDirector | None
+    characterStream: _DOMInputSourceCharacterStreamType
+    stringData: _DOMInputSourceStringDataType
+    encoding: _DOMInputSourceEncodingType
+    publicId: str | None
+    systemId: str | None
+    baseURI: str | None
+    def _get_byteStream(self) -> OpenerDirector | None: ...
+    def _set_byteStream(self, byteStream: OpenerDirector | None) -> None: ...
+    def _get_characterStream(self) -> _DOMInputSourceCharacterStreamType: ...
+    def _set_characterStream(self, characterStream: _DOMInputSourceCharacterStreamType) -> None: ...
+    def _get_stringData(self) -> _DOMInputSourceStringDataType: ...
+    def _set_stringData(self, data: _DOMInputSourceStringDataType) -> None: ...
+    def _get_encoding(self) -> _DOMInputSourceEncodingType: ...
+    def _set_encoding(self, encoding: _DOMInputSourceEncodingType) -> None: ...
+    def _get_publicId(self) -> str | None: ...
+    def _set_publicId(self, publicId: str | None) -> None: ...
+    def _get_systemId(self) -> str | None: ...
+    def _set_systemId(self, systemId: str | None) -> None: ...
+    def _get_baseURI(self) -> str | None: ...
+    def _set_baseURI(self, uri: str | None) -> None: ...
```

Comment:
There's still quite a lot of `_non_public` methods, and also a few attributes in `DOMBuilder`.